### ISSUE TITLE
Controller 100% completely doesn't work

### ISSUE THREAD

**Issue by spamzilla** [HarbourMasters/Starship#51]
My controller 100% completely does not work. It is a PlayStation 2 controller with a crappy USB adapter. It has issues with some programs but never this bad! It usually registers as a PS3 controller actually.

I just noticed this line in my log:

`[18:26:46.022] [os.cpp:27] [error] Failed add SDL game controller mappings from "./gamecontrollerdb.txt" (Invalid RWops)`

**Comment by BuckoA511** [comment]
I don't think the game supports directinput controllers at all.

**Comment by Calinou** [comment]
Regarding DInput, see https://github.com/HarbourMasters/Starship/issues/54.

The Windows release seems to be missing `gamecontrollerdb.txt` like 2Ship2Harkinian has. You could try copying it to the Starship folder but it didn't work on a ROG Ally X for me (nor an external Xbox One V3 controller).

**Comment by spamzilla** [comment]
So I copied `gamecontrollerdb.txt` into my root folder and the controller was then displayed. The buttons were really out-of-whack but I did get an acceptable configuration to play with. There seems to be another problem now though. It appears that one unknown button is continually being pressed. This is evident on the map screen where the camera is forced down completely and I can't get it to move back up. I think it is maybe not a big deal as it isn't affecting gameplay. I was able to beat both sides of Venom and it was a good time.
